src/Api/Api.php

```diff
@@ -91,10 +92,15 @@ public function run(string $source, array $params): string
 
         foreach ($this->manipulators as $manipulator) {
             $manipulator->setParams($params);
-
+            /** @var \Intervention\Image\Interfaces\ImageInterface */
             $image = $manipulator->run($image);
         }
 
-        return $image->encodeByMediaType()->toString();
+        $encoder = new Encode();
+        $encoder->setParams($params);
+        /** @var \Intervention\Image\Interfaces\EncodedImageInterface */
+        $encoded = $encoder->run($image);
+
+        return $encoded->toString();
```

Comment:
Since there was some important changes on how Intervention process images, it was necessary to move the Encode manipulator to run here instead. 

This will make sure that it will run at the end of all manipulators, sine it returns a `EncodedImage` instance. Also it will avoid the strategy used before, causing double encoding of the given image.

Reply:
Ok, makes sense.

I moved the Encoder file to the Api folder, just to keep SoC and not pollute Api.php file.